commit dc12027209380215e94118463f1ae3ecdf409a6b
Author: Daniel Firth <dan.firth@homotopic.tech>
Date:   Sun Dec 3 09:15:18 2023 +0000

    hide unzip

diff --git a/library/HTMLEntities/Prelude.hs b/library/HTMLEntities/Prelude.hs
index a8b82de..9b3d9c9 100644
--- a/library/HTMLEntities/Prelude.hs
+++ b/library/HTMLEntities/Prelude.hs
@@ -25,7 +25,7 @@ import Data.Either as Exports
 import Data.Fixed as Exports
 import Data.Foldable as Exports hiding (toList)
 import Data.Function as Exports hiding (id, (.))
-import Data.Functor as Exports
+import Data.Functor as Exports hiding (unzip)
 import Data.Functor.Compose as Exports
 import Data.IORef as Exports
 import Data.Int as Exports
